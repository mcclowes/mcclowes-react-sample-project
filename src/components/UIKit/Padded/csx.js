import styled from "styled-components/macro";

const spaces = {
  0: "0",
  1: "0.125em",
  2: "0.25em",
  3: "0.5em",
  4: "0.75em",
  5: "1em",
};

const getDirection = (direction) => {
  if (direction === "t") return "-top";
  if (direction === "b") return "-bottom";
  if (direction === "l") return "-left";
  if (direction === "r") return "-right";
  return "";
};

const getSpacing = (space) => {
  const [padding, amount] = space.match(/[a-z]+|[^a-z]+/gi);

  const spaceType = padding.split("")[0] === "p" ? "padding" : "margin";
  const direction = getDirection(padding.split("")[1]);

  return spaceType + direction + ": " + spaces[amount];
};

export const PaddedWrapper = styled.div`
  ${(props) => props.space && getSpacing(props.space)};
`;
