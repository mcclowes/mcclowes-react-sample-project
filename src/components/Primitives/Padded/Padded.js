import PropTypes from "prop-types";
import React from "react";
import { PaddedWrapper } from "./csx";

const Padded = (props) => {
  const { space = "p1", children } = props;

  return <PaddedWrapper space={space}>{children}</PaddedWrapper>;
};

Padded.propTypes = {
  children: PropTypes.any,
  space: PropTypes.string,
};

export default Padded;
