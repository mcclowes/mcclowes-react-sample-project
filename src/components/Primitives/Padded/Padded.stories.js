import React from "react";
import Padded from "./Padded";

export default { title: "Padded" };

export const simple = () => {
  return <Padded>Some padded text</Padded>;
};

export const nested = () => {
  return (
    <Padded>
      <Padded>
        <Padded>Super nested</Padded>
      </Padded>
    </Padded>
  );
};

export const p1 = () => {
  return <Padded space="p1">Some padded text</Padded>;
};
export const p2 = () => {
  return <Padded space="p2">Some padded text</Padded>;
};
export const p3 = () => {
  return <Padded space="p3">Some padded text</Padded>;
};
export const p4 = () => {
  return <Padded space="p4">Some padded text</Padded>;
};
export const p5 = () => {
  return <Padded space="p5">Some padded text</Padded>;
};
