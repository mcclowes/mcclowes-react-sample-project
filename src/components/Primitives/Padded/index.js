import Padded from "./Padded";

export default Padded;
