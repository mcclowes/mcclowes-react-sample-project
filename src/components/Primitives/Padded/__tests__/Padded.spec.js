import Padded from "../Padded";
import React from "react";
import { shallow } from "enzyme";

const defaultProps = {};

const renderComponent = (newProps) => {
  const props = {
    ...defaultProps,
    ...newProps,
  };

  return shallow(<Padded {...props} />);
};

let wrapper;

describe("Padded", () => {
  it("renders default component", () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  it("renders correct padding", () => {
    expect(renderComponent({ space: "p0" })).toMatchSnapshot();
    expect(renderComponent({ space: "pr1" })).toMatchSnapshot();
    expect(renderComponent({ space: "p2" })).toMatchSnapshot();
    expect(renderComponent({ space: "pt3" })).toMatchSnapshot();
    expect(renderComponent({ space: "pb4" })).toMatchSnapshot();
    expect(renderComponent({ space: "pl5" })).toMatchSnapshot();
  });

  it("renders correct margin", () => {
    expect(renderComponent({ space: "m0" })).toMatchSnapshot();
    expect(renderComponent({ space: "mr1" })).toMatchSnapshot();
    expect(renderComponent({ space: "m2" })).toMatchSnapshot();
    expect(renderComponent({ space: "mt3" })).toMatchSnapshot();
    expect(renderComponent({ space: "mb4" })).toMatchSnapshot();
    expect(renderComponent({ space: "ml5" })).toMatchSnapshot();
  });
});
