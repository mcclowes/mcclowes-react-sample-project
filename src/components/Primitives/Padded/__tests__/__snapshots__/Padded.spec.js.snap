// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Padded renders correct margin 1`] = `
<csx__PaddedWrapper
  space="m0"
/>
`;

exports[`Padded renders correct margin 2`] = `
<csx__PaddedWrapper
  space="mr1"
/>
`;

exports[`Padded renders correct margin 3`] = `
<csx__PaddedWrapper
  space="m2"
/>
`;

exports[`Padded renders correct margin 4`] = `
<csx__PaddedWrapper
  space="mt3"
/>
`;

exports[`Padded renders correct margin 5`] = `
<csx__PaddedWrapper
  space="mb4"
/>
`;

exports[`Padded renders correct margin 6`] = `
<csx__PaddedWrapper
  space="ml5"
/>
`;

exports[`Padded renders correct padding 1`] = `
<csx__PaddedWrapper
  space="p0"
/>
`;

exports[`Padded renders correct padding 2`] = `
<csx__PaddedWrapper
  space="pr1"
/>
`;

exports[`Padded renders correct padding 3`] = `
<csx__PaddedWrapper
  space="p2"
/>
`;

exports[`Padded renders correct padding 4`] = `
<csx__PaddedWrapper
  space="pt3"
/>
`;

exports[`Padded renders correct padding 5`] = `
<csx__PaddedWrapper
  space="pb4"
/>
`;

exports[`Padded renders correct padding 6`] = `
<csx__PaddedWrapper
  space="pl5"
/>
`;

exports[`Padded renders default component 1`] = `
<csx__PaddedWrapper
  space="p1"
/>
`;
